### ISSUE TITLE
sendto failed: EPIPE (Broken pipe)

### ISSUE THREAD

**Issue by jjhesk** [square/retrofit#2049]
Followed by the previous thread #2013 and [OKHTTP issue](https://github.com/square/okhttp/issues/1775) .

Are there possible solution to resolve this? If @swankjesse mentioned that this is possible the connection issue during uploading. From my testing reports, there are evidence showing that this crash EPIPE is way before the server is start noticing the request from the client side. This lead to possible error from reading the file internally. Noticed from the console monitor it shown that these items are recorded.

```
....   !EOF
-----> beginning of crash
```
I am thinking if this is related to possible leak from reading datastream.

To reproduce this error, I have prepared a 6.4MB of image file. With the following implementation. 
RETROFIT LIBRARY=2.1.0
OKHTTP LIBRARY=3.3.1

Interface of upload
```java
 private interface UploadTest {
 @POST("uploadtest/{test_id}/")
  Call<ResBool> doUpload(@Path("test_id") final String testId, @Body RequestBody mfile);
}
```

In the main operation we have.
```java

private Call<ResBool> singlequeueuploadcall;

void request(){
 singlequeueuploadcall = requestbuilder.uploadmapbody_adv(getInterceptor().getUserIdOnly(), preparebody(upload_subject)).clone();
singlequeueuploadcall.enqueue(new Callback<ResBool>() {
    ....
```

For the ` requestbuilder`
```java



        header_control = new RestAdapter(app);
        OkHttpClient.Builder cb = new OkHttpClient.Builder();
        cb.addInterceptor(header_control);
        cb.connectTimeout(5, TimeUnit.MINUTES);
        cb.readTimeout(5, TimeUnit.MINUTES);
        final HttpLoggingInterceptor logger = new HttpLoggingInterceptor();
        logger.setLevel(HttpLoggingInterceptor.Level.BODY);
        cb.addNetworkInterceptor(logger);
        OkHttpClient client =  cb.build();
 

    Retrofit retrofit_done = new Retrofit.Builder()
                .baseUrl(api_endpoint)
                .client(client)
                .build();



UploadTest requestbuilder = retrofit_done.create(UploadTest.class)
```

To defined the `RequestBody` from using `preparebody`
```java
   private RequestBody preparebody(File photo) {
        RequestBody body = new MultipartBody.Builder()
                .setType(MultipartBody.FORM)
                //.addFormDataPart("name", name)
                .addFormDataPart("art", photo.getName(), RequestBody.create(MediaType.parse("image/jpeg"), photo))
                .build();


        try {
            long size = body.contentLength();
            MediaType type = body.contentType();
        } catch (IOException e) {
            e.printStackTrace();
        }

        return body;
    }


}
```

Basically the setup for the request body is like this.

```java
   RequestBody body = new MultipartBody.Builder()
                .setType(MultipartBody.FORM)
                //.addFormDataPart("name", name)
                .addFormDataPart("art", photo.getName(), RequestBody.create(MediaType.parse("image/jpeg"), photo))
                .build();
```

**Comment by JakeWharton** [comment]
This crash happens far below Retrofit. We only ask the HTTP client to do stuff and the crash happens even below the HTTP client. No action to take here.

**Comment by jjhesk** [comment]
@JakeWharton Gread God -> where should I looking to? native C++ level?
